client/my-sites/themes/theme-showcase.jsx

```diff
@@ -0,0 +1,117 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import pickBy from 'lodash/pickBy';
+import merge from 'lodash/merge';
+import get from 'lodash/get';
+
+/**
+ * Internal dependencies
+ */
+import Main from 'components/main';
+import ThemePreview from './theme-preview';
+import ThemesSelection from './themes-selection';
+import PageViewTracker from 'lib/analytics/page-view-tracker';
+import { addTracking } from './helpers';
+import actionLabels from './action-labels';
+
+const ThemeShowcase = React.createClass( {
+	propTypes: {
+		// Connected props
+		options: PropTypes.objectOf( PropTypes.shape( {
+			label: PropTypes.string.isRequired,
+			header: PropTypes.string,
+			getUrl: PropTypes.func,
+			action: PropTypes.func
+		} ) ),
+		defaultOption: PropTypes.string,
+		getScreenshotOption: PropTypes.func
+	},
+
+	getDefaultProps() {
+		return {
+			selectedSite: false
+		};
+	},
+
+	getInitialState() {
+		return {
+			showPreview: false,
+			previewingTheme: null,
+		};
+	},
+
+	togglePreview( theme ) {
+		this.setState( { showPreview: ! this.state.showPreview, previewingTheme: theme } );
+	},
+
+	onPreviewButtonClick( theme ) {
+		const defaultOption = this.props.options[ this.props.defaultOption ];
+		this.setState( { showPreview: false }, () => {
+			defaultOption.action( theme );
+		} );
+	},
+
+	render() {
+		// If a preview action is passed, use that. Otherwise, use our own.
+		const previewAction = get(
+			this.props.options,
+			[ 'preview', 'action' ],
+			theme => this.togglePreview( theme )
+		),
+			buttonOptions = merge(
+				{},
+				this.props.options,
+				{ preview: {
+					label: actionLabels.preview.label,
+					action: previewAction
+				} }
+			),
+			defaultOption = this.props.options[ this.props.defaultOption ],
+			getScreenshotOption = theme => buttonOptions[ this.props.getScreenshotOption( theme ) ];
```

Comment:
similar to the above, `getScreenshotOption` isn't required, and even it it were, the reference could produce `null` if the object property doesn't exist.